modyn/trainer_server/internal/trainer/pytorch_trainer.py

```diff
@@ -418,6 +433,66 @@ def train(self) -> None:  # pylint: disable=too-many-locals, too-many-branches
                         stopw.start("DownsampleBTS", resume=True)
                         data, sample_ids, target, weights = self.downsample_batch(data, sample_ids, target)
                         stopw.stop()
+                        assert (
+                            all(tensor.shape[0] == expected_bts_size for tensor in data.values())
+                            if isinstance(data, dict)
+                            else data.shape[0] == expected_bts_size
+                        ), (
+                            f"expected bts size: {expected_bts_size} "
+                            + f"actual: {data.shape[0] if isinstance(data, torch.Tensor) else 'n/a'}"
+                        )
+
+                        assert (
+                            len(sample_ids) == expected_bts_size
+                        ), f"expected bts size: {expected_bts_size} actual: {len(sample_ids)}"
+                        assert (
+                            target.shape[0] == expected_bts_size
+                        ), f"expected bts size: {expected_bts_size} actual: {target.shape[0]}"
```

Comment:
I will move the assertions to a separate function to debloat `train`, but we need them because the output size here depends on the downsamplers outputting the expected number of items. Since the trainer cannot force them to, I'd rather assert this here